src/test/codegen/remap_path_prefix/issue-73167-remap-std.rs

```diff
@@ -0,0 +1,16 @@
+// ignore-windows
+// ignore-tidy-linelength
+
+// compile-flags: -g  -C no-prepopulate-passes --remap-path-prefix=/home/USERNAME=REDACTED
+// aux-build:pseudo_std.rs
+
+extern crate pseudo_std;
+// Here we check that imported code from std has their path remapped.
+// If we were to use an actual std function to test, we'd have to remap {{cwd}}/library/...,
+// but that will break if rustc was built with remap-debuginfo = true in config.toml.
+// Hence we simulate the path rust-src would take in a real linux installation
+// by remapping the prefix inside pseudo_std.rs
+// CHECK: !DIFile(filename: "REDACTED/TOOLCHAIN/lib/rustlib/src/rust/library/std/src/pseudo_std.rs", directory: ""
+fn main() {
+    pseudo_std::some_std_function();
+}
```

Comment:
Ideally this could be a single test file with the exact code snippet from #73167: https://github.com/rust-lang/rust/pull/83813/commits/d760af1c308400c967843ce04384d73f3c8c4b10. Instead of this workaround it might be more proper to add a header command to ignore it when `remap-debuginfo = true`, though I'm not sure.

Reply:
While this test does fail with today's rustc and is therefore useful, I'm not entirely sure this hits the code paths that are taken when the `rust-src` issue comes up. In particular the codepaths related to the `CFG_VIRTUAL_RUST_SOURCE_BASE_DIR` environment variable are not invoked.

As `CFG_VIRTUAL_RUST_SOURCE_BASE_DIR` is set by bootstrap to a value computed by this snippet… https://github.com/rust-lang/rust/blob/97717a561844eccbb6d6cc114adb94a8fa4e0172/src/bootstrap/lib.rs#L821-L824

…so I think we could write a makefile test that builds a `pseudo_std` a lot more like `std` or other sysroot crates with respect to path remapping behaviour.

---

Did the test you added in the first commit not work right? Why?

Reply:
The test in the first commit works when `remap-debuginfo = false` in `config.toml`, but it breaks when it's set to true because `/rustc/$version/library...` will be imported rather than `{{cwd}}/library`, so remapping from `{{cwd}}` to something else won't do anything.

I think it's better to revert to the original single-file test, and use a `CHECK-NOT:` on `{{cwd}}` instead. This way the behaviour of having `rust-src` would be correctly simulated and tested when `remap-debuginfo = false`, but the test would still pass with `remap-debuginfo = true`.

Reply:
The plan seems fine, though I still think we should keep both tests as they still exercise slightly different code paths.

Reply:
I renamed the new test and added back in the test in the first commit. It turns out that I can't use `{{cwd}}` inside a `CHECK-NOT:` directive (it'll be preserved literally and not expanded) and FileCheck itself doesn't provide similar environment variables, so I used regex wildcard on what would've been `{{cwd}}` and made it work on the path after that instead (`{{cwd}}/library/std` getting remapped to something else without the `library/std` bit)